sql/catalog/lease: TestLeaseAtLatestVersion failed

sql/catalog/lease.TestLeaseAtLatestVersion [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20947820?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20947820?buildTab=artifacts#/) on release-26.1 @ [8d9b02ad24eaab27e40e294088fdec73139848a9](https://github.com/cockroachdb/cockroach/commits/8d9b02ad24eaab27e40e294088fdec73139848a9):


```
=== RUN   TestLeaseAtLatestVersion
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/1e31bde6494190748168afe7a747530a/logTestLeaseAtLatestVersion2801178543
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:182: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:668: DRPC is enabled (via override by TestingGlobalDRPCOption)
    lease_test.go:1243: not seeing latest version
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1e31bde6494190748168afe7a747530a/logTestLeaseAtLatestVersion2801178543
--- FAIL: TestLeaseAtLatestVersion (1.74s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseAtLatestVersion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58487

--- a/pkg/sql/catalog/lease/lease_test.go
+++ b/pkg/sql/catalog/lease/lease_test.go
@@ -1170,16 +1170,23 @@ func TestLeaseAtLatestVersion(t *testing.T) {
 
 	var params base.TestServerArgs
 	errChan := make(chan error, 1)
+	// Use an atomic to track when Publish has completed. We only care about
+	// lease acquisitions that happen after Publish, since the WaitForInitialVersion
+	// feature may trigger an automatic acquisition of version 1 during CREATE TABLE.
+	var afterPublish atomic.Bool
 	params.Knobs = base.TestingKnobs{
 		SQLLeaseManager: &lease.ManagerTestingKnobs{
 			LeaseStoreTestingKnobs: lease.StorageTestingKnobs{
 				LeaseAcquiredEvent: func(desc catalog.Descriptor, _ error) {
-					if desc.GetName() == "kv" {
+					if desc.GetName() == "kv" && afterPublish.Load() {
 						var err error
 						if desc.GetVersion() < 2 {
 							err = errors.Errorf("not seeing latest version")
 						}
-						errChan <- err
+						select {
+						case errChan <- err:
+						default:
+						}
 					}
 				},
 			},
@@ -1225,6 +1232,7 @@ INSERT INTO t.sc1.timestamp VALUES ('a', 'b');
 				}, nil); err != nil {
 				t.Fatal(err)
 			}
+			afterPublish.Store(true)
 			updated = true
 		}
 		// This select will see version 1 of the table. It will first

